contracts/libraries/UserLib.sol

```diff
@@ -177,66 +191,87 @@ library UserLib {
+      updateUserRating(userContext, usersRating[i].user, usersRating[i].rating, communityId);
     }
   }
 
   /// @notice Add rating to user
   /// @param userAddr user's rating will be change
   /// @param rating value for add to user's rating
-  function updateUserRating(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating) internal {
+  function updateUserRating(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating, uint32 communityId) internal {
     if (rating == 0) return;
 
-    updateRatingBase(userContext, user, userAddr, rating);
+    updateRatingBase(userContext, user, userAddr, rating, communityId);
   }
 
-  function updateUserRating(UserLib.UserContext storage userContext, address userAddr, int32 rating) internal {
+  function updateUserRating(UserLib.UserContext storage userContext, address userAddr, int32 rating, uint32 communityId) internal {
     if (rating == 0) return;
 
     User storage user = getUserByAddress(userContext.users, userAddr);
-    updateRatingBase(userContext, user, userAddr, rating);
+    updateRatingBase(userContext, user, userAddr, rating, communityId);
   }
 
-  function updateRatingBase(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating) internal {
+  function updateRatingBase(UserLib.UserContext storage userContext, User storage user, address userAddr, int32 rating, uint32 communityId) internal {
     uint16 currentPeriod = RewardLib.getPeriod(CommonLib.getTimestamp());
-    // User storage user = getUserByAddress(userContext.users, userAddr);
-    int32 newRating = user.rating += rating;
-    uint256 pastPeriodsCount = user.rewardPeriods.length;
     
-    RewardLib.PeriodRating storage currentWeekRating = RewardLib.getUserPeriodRating(userContext.userRewards, userAddr, currentPeriod);
-    bool isFirstTransactionOnThisWeek = pastPeriodsCount == 0 || user.rewardPeriods[pastPeriodsCount - 1] != currentPeriod; 
+    CommunityRatingForUser storage communityUser = userContext.userRatingCollection.communityRatingForUser[userAddr];
+    UserRating storage userRating = communityUser.userRating[communityId];
+    if (!userRating.isActive) {
+      userRating.rating = START_USER_RATING;
+      userRating.payOutRating = START_USER_RATING;
+      userRating.isActive = true;
+    }
+
+    int32 newRating = userRating.rating += rating;
+    uint256 pastPeriodsCount = communityUser.rewardPeriods.length;
+
+    RewardLib.PeriodRating storage currentWeekRating = RewardLib.getUserPeriodRating(communityUser.userRewards[currentPeriod], communityId);
+    bool isFirstTransactionOnThisWeek = pastPeriodsCount == 0 || communityUser.rewardPeriods[pastPeriodsCount - 1] != currentPeriod; 
     if (isFirstTransactionOnThisWeek) {
-      user.rewardPeriods.push(currentPeriod);
+      communityUser.rewardPeriods.push(currentPeriod);
+
+      bool isActiveInCommunity;  // name
```

Comment:
I don't like that we have to read so many values every time when updating the rating for the user.

Reply:
Should it check if the community needs to be added to the array even if `isFirstTransactionOnThisWeek` is `false`?